### PR TITLE
Issue 105353: Add messages+details to audit logs response

### LINKED ISSUE #105353: Capture the status message from status-type responses in audit logs
#### What happened:

If the response to a request is a `metav1.Status`, we currently copy a subset of the response to the `ResponseStatus` field of the audit logs: https://github.com/kubernetes/kubernetes/blob/1ad3e14f1fc4a5f6e6cd6f9ffc7d030cb3d6540d/staging/src/k8s.io/apiserver/pkg/audit/request.go#L183-L188

Note that the message is not included. In order to capture the response message, the full `RequestResponse` audit level must be used. This is not always desireable - as a contrived example, consider the case of secret resource failing validation. For a more common case, dropping the full response object may be desireable to cut down on storage costs, but the relatively small status response still holds valuable data (e.g. admission denials) in the error case.

#### What you expected to happen:

We should always capture the full `metav1.Status` response, including the message field. I'm open to dropping the details field, but I'm not sure that's necessary.

/sig auth api-machinery
/area audit
/milestone v1.23

### PR DESCRIPTION
**Pull request by uhari03**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
It was noted in the issue that the message and details were missing from the audit logs. This PR addresses that.

#### Which issue(s) this PR fixes:

Fixes #105353

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by tallclair — COMMENTED** [review]
Thanks!

**Comment by enj** [comment]
/milestone v1.23

@uhari03 code freeze for v1.23 is in a week.  Please address Tim's comments ASAP if you would like to have this go into the next release.

**Comment by uhari03** [comment]
Will get to this tonight after work - apologies for the delay

**Comment by uhari03** [comment]
@tallclair @enj Sorry for the wait, I hope it's good now

**Comment by tallclair** [comment]
Please run `./hack/update-gofmt.sh` to pass verify.

**Comment by uhari03** [comment]
Honestly, not too sure why my CLA is failing all of a sudden.

**Comment by uhari03** [comment]
Thanks @WillsonHG

**Comment by reylejano** [comment]
/milestone clear
Release Managers are cutting 1.23.0-rc.0 and creating the `release-1.23` branch

**Comment by uhari03** [comment]
@tallclair Lmk if it looks good now

**Comment by uhari03** [comment]
@tallclair Bump on this when you can please

**Comment by dims** [comment]
@uhari03 can you please squash the commits into one? (one line change should not have 4 commits)

**Comment by uhari03** [comment]
@dims lmk if it's ok now

**Comment by enj** [comment]
/milestone v1.24

@tallclair PTAL

File: staging/src/k8s.io/apiserver/pkg/audit/request.go [hunk 1]
```diff
@@ -195,11 +195,7 @@ func LogResponseObject(ctx context.Context, obj runtime.Object, gv schema.GroupV
 	}
 	if status, ok := obj.(*metav1.Status); ok {
 		// selectively copy the bounded fields.
-		ae.ResponseStatus = &metav1.Status{
-			Status: status.Status,
-			Reason: status.Reason,
-			Code:   status.Code,
-		}
+		ae.ResponseStatus = status
```
Comment (tallclair) [inline staging/src/k8s.io/apiserver/pkg/audit/request.go:198]: I took a closer look at what's included in a `metav1.Status`, and I think there's some stuff we'd want to filter out. Notably, the `ListMeta` is just going to be noise that bloats the logs. I'd suggest falling back to the previous approach, something like:

```suggestion
		ae.ResponseStatus = &metav1.Status{
			Status: status.Status,
			Message: status.Message,
			Reason: status.Reason,
			Details: status.Details,
			Code:   status.Code,
		}
```

I don't have an opinion on whether to include details.

Comment (uhari03) [reply staging/src/k8s.io/apiserver/pkg/audit/request.go]: @tallclair please take a look again when you get the chance please

**Pull request merged**